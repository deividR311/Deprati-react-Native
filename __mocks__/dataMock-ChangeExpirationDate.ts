export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
