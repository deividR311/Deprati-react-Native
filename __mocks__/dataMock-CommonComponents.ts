export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
