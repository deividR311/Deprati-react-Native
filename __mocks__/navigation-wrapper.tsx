import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
