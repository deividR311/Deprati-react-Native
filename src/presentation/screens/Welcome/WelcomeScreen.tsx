//Libs
import React, { useLayoutEffect } from 'react';
import { SafeAreaView, ScrollView, View } from 'react-native';
//hooks
import { useNavigation } from '@react-navigation/native';
import { useEmmaSdk } from '../../../infrastucture/native-modules/emma';
//components
import { ButtonText, MainButton } from '../../common-components/buttons/Button';
import { SkeletonWelcome } from './skeletonWelcome';
import ExtensionComponent from '../../common-components/extension-component';
import ErrorPage from '../ErrorPage';
//Styles
import { styles } from './stylesWelcome';
//Utils
import { NAV } from '../../../application/common/namesNav';
import { keyEvents } from '../../../infrastucture/native-modules/emma/clickEventMap';
import layout from '../../../application/common/layout';
import usePageContent from '../../../infrastucture/apis/contentPage/contentPage.hook';

export default function WelcomeScreen() {
  const navigation = useNavigation();
  const { trackEventclick } = useEmmaSdk({});

  const { loading, error, pageContent, getDataContent } = usePageContent();

  useLayoutEffect(() => {
    getDataContent({
      pageLabelOrId: 'welcomePage'
    });
  }, []);

  const handleGoTo = (nav: string) => {
    switch (nav) {
      case NAV.AUTH_NAVIGATION:
        navigation.navigate(nav as never);
        break;
      case NAV.DASHBOARD_NAVIGATION:
        navigation.replace(nav);
        break;
    }
  };

  if (error) return <ErrorPage />;
  if (loading) return <SkeletonWelcome />;
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView
        contentContainerStyle={[
          styles.contentContainer,
          !layout.isSmallDevice && styles.heightAux
        ]}>
        <View style={styles.containerImage}>
          <View style={styles.image}>
            {pageContent?.components?.SiteLogoComponent && (
              <ExtensionComponent
                {...pageContent.components.SiteLogoComponent}
              />
            )}
          </View>
        </View>
        <View style={styles.containerCarousel}>
          {pageContent?.components?.DePratiWelcomeRotatingImagesComponent && (
            <ExtensionComponent
              {...pageContent?.components
                ?.DePratiWelcomeRotatingImagesComponent}
            />
          )}
        </View>
        <View style={styles.containerButtons} accessible>
          <MainButton
            testID="welcome_iniciar_sesion"
            title="INICIAR SESIÃ“N"
            onPress={() => {
              handleGoTo(NAV.AUTH_NAVIGATION);
            }}
          />
          <ButtonText
            testID="welcome_btn_omitir"
            onPress={() => {
              trackEventclick(keyEvents.prelogin_saltar);
              handleGoTo(NAV.DASHBOARD_NAVIGATION);
            }}
            title={
              pageContent?.components?.SkipWelcomeLink?.linkName ?? 'Saltar'
            }
          />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}
