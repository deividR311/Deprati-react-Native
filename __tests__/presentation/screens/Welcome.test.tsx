import React from 'react';
import { render } from '@testing-library/react-native';
import WelcomeScreen from '../../../src/presentation/screens/Welcome/WelcomeScreen';
import { NavigationWrapper } from '../../../__mocks__/navigation-wrapper';
const dirMockHook =
  '../../../src/infrastucture/apis/contentPage/contentPage.hook';

describe('Welcome', () => {
  it('renders correctly in Snapshot', () => {
    jest.spyOn(require(dirMockHook), 'usePageContent').mockReturnValueOnce({
      loading: false,
      pageContent: {
        defaultPage: true,
        name: 'WelcomePage',
        template: 'WelcomePageTemplate',
        typeCode: 'ContentPage',
        uid: 'welcomePage',
        timestamp: 1679610740894,
        components: {
          DePratiWelcomeRotatingImagesComponent: {
            uid: 'DePratiWelcomeRotatingImagesComponent',
            typeCode: 'DepratiWelcomeRotatingImagesComponent',
            modifiedTime: '2022-04-08T10:39:02.55-05:00',
            name: 'Deprati Welcome Rotating Images Component',
            container: 'false',
            effect: 'FADE',
            rotatingImagesComponentsList: 'DePratiWelcomeCarousel_Mobile',
            childrenComponentsList: ['DePratiWelcomeCarousel_Mobile'],
            childrenComponents: {
              DePratiWelcomeCarousel_Mobile: {
                uid: 'DePratiWelcomeCarousel_Mobile',
                typeCode: 'RotatingImagesComponent',
                modifiedTime: '2022-04-08T15:38:40+0000',
                name: 'DePrati Homepage Mobile Carousel',
                container: 'false',
                resolutionEnum: 'MOBILE',
                effect: 'FADE',
                banners:
                  'WelcomeCarousel_1_Mobile WelcomeCarousel_2_Mobile WelcomeCarousel_3_Mobile',
                timeout: '6',
                childrenComponentsList: [
                  'WelcomeCarousel_1_Mobile',
                  'WelcomeCarousel_2_Mobile',
                  'WelcomeCarousel_3_Mobile'
                ],
                childrenComponents: {
                  WelcomeCarousel_1_Mobile: {
                    uid: 'WelcomeCarousel_1_Mobile',
                    typeCode: 'BannerComponent',
                    modifiedTime: '2022-05-18T21:25:31+0000',
                    name: 'Example1 Home Banner Mobile',
                    container: 'false',
                    external: 'false',
                    media: {
                      code: 'WelcomeCarousel_1_Mobile.png',
                      mime: 'image/png',
                      url: 'https://imagestg.deprati.com.ec/sys-master/images/h6f/h47/8969902358558',
                      downloadUrl:
                        'https://imagestg.deprati.com.ec/sys-master/images/h6f/h47/8969902358558'
                    },
                    headline: 'Bienvenido A De Prati App - Prueba',
                    content: 'Prueba iPhone - texto texto texto',
                    urlLink: '/'
                  },
                  WelcomeCarousel_2_Mobile: {
                    uid: 'WelcomeCarousel_2_Mobile',
                    typeCode: 'BannerComponent',
                    modifiedTime: '2023-01-25T14:25:41+0000',
                    name: 'Example2 Home Banner Mobile',
                    container: 'false',
                    external: 'false',
                    media: {
                      code: 'WelcomeCarousel_2_Mobile.png',
                      mime: 'image/png',
                      url: 'https://imagestg.deprati.com.ec/sys-master/images/h8e/he8/8969829449758',
                      downloadUrl:
                        'https://imagestg.deprati.com.ec/sys-master/images/h8e/he8/8969829449758'
                    },
                    headline: 'Encuentra tus productos favoritos',
                    content:
                      'Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit',
                    urlLink: '/'
                  },
                  WelcomeCarousel_3_Mobile: {
                    uid: 'WelcomeCarousel_3_Mobile',
                    typeCode: 'BannerComponent',
                    modifiedTime: '2023-01-25T14:25:41+0000',
                    name: 'Example3 Home Banner Mobile',
                    container: 'false',
                    external: 'false',
                    media: {
                      code: 'WelcomeCarousel_3_Mobile.png',
                      mime: 'image/png',
                      url: 'https://imagestg.deprati.com.ec/sys-master/images/h23/he8/8969829482526',
                      downloadUrl:
                        'https://imagestg.deprati.com.ec/sys-master/images/h23/he8/8969829482526'
                    },
                    headline:
                      'Solicita tu Crédito De Prati de forma facíl y rápida - Prueba iPhone',
                    content:
                      'Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit. Hola.',
                    urlLink: '/'
                  }
                }
              }
            }
          },
          SkipWelcomeLink: {
            uid: 'SkipWelcomeLink',
            typeCode: 'CMSLinkComponent',
            modifiedTime: '2022-06-22T12:28:34.64-05:00',
            name: 'SkipWelcomeLink',
            container: 'false',
            external: 'false',
            linkName: 'Invitado',
            url: 'homePage',
            target: 'false',
            childrenComponentsList: null,
            childrenComponents: {}
          },
          SiteLogoComponent: {
            uid: 'SiteLogoComponent',
            typeCode: 'SimpleBannerComponent',
            modifiedTime: '2022-05-20T11:47:01.945-05:00',
            name: 'Site Logo Component',
            container: 'false',
            external: 'false',
            media: {
              code: '/images/logo_deprati_responsive.svg',
              mime: 'image/svg+xml',
              altText: 'Deprati Storefront',
              url: 'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502',
              downloadUrl:
                'https://imagestg.deprati.com.ec/sys-master/images/h9f/hfd/8969934569502'
            },
            urlLink: '/',
            childrenComponentsList: null,
            childrenComponents: {}
          }
        },
        slots: {
          Section1: ['DePratiWelcomeRotatingImagesComponent'],
          Section2: ['SkipWelcomeLink'],
          SiteLogo: ['SiteLogoComponent']
        }
      },
      getDataContent: jest.fn(),
      error: false
    });
    const screenTest = render(<NavigationWrapper screen={WelcomeScreen} />);
    const { getByTestId } = screenTest;
    expect(getByTestId('welcome_iniciar_sesion')).toBeTruthy();
    expect(getByTestId('welcome_btn_omitir')).toBeTruthy();
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  it('renders correctly in Loadind Skeleton', () => {
    jest.spyOn(require(dirMockHook), 'usePageContent').mockReturnValueOnce({
      loading: true,
      pageContent: {},
      getDataContent: jest.fn(),
      error: false
    });
    const { findByTestId } = render(
      <NavigationWrapper screen={WelcomeScreen} />
    );
    expect(findByTestId('skeletonSignIn')).toBeTruthy();
  });

  it('renders correctly in Error Skeleton', () => {
    jest.spyOn(require(dirMockHook), 'usePageContent').mockReturnValueOnce({
      error: true,
      loading: false,
      getDataContent: jest.fn()
    });
    const { findByTestId } = render(
      <NavigationWrapper screen={WelcomeScreen} />
    );
    expect(findByTestId('errorPage')).toBeTruthy();
  });
});
