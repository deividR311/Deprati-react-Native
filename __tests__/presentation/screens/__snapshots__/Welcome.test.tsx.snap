// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "#ffffff",
            }
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Array [
                Object {
                  "paddingBottom": 0,
                },
                Object {
                  "height": 1334,
                },
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": "10%",
                    "paddingBottom": 16,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "height": "auto",
                            "marginTop": 5,
                            "width": "100%",
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 5,
                    "elevation": 8,
                    "height": "60%",
                    "shadowColor": "rgba(16, 10, 68, 0.4)",
                    "shadowOffset": Object {
                      "height": 8,
                      "width": 4,
                    },
                    "shadowOpacity": 0.4,
                    "shadowRadius": 20,
                    "width": "90%",
                  }
                }
              >
                <RCTScrollView
                  collapsable={false}
                  data={
                    Array [
                      Object {
                        "container": "false",
                        "content": "Prueba iPhone - texto texto texto",
                        "external": "false",
                        "headline": "Bienvenido A De Prati App - Prueba",
                        "media": Object {
                          "code": "WelcomeCarousel_1_Mobile.png",
                          "downloadUrl": "https://imagestg.deprati.com.ec/sys-master/images/h6f/h47/8969902358558",
                          "mime": "image/png",
                          "url": "https://imagestg.deprati.com.ec/sys-master/images/h6f/h47/8969902358558",
                        },
                        "modifiedTime": "2022-05-18T21:25:31+0000",
                        "name": "Example1 Home Banner Mobile",
                        "typeCode": "BannerComponent",
                        "uid": "WelcomeCarousel_1_Mobile",
                        "urlLink": "/",
                      },
                      Object {
                        "container": "false",
                        "content": "Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit",
                        "external": "false",
                        "headline": "Encuentra tus productos favoritos",
                        "media": Object {
                          "code": "WelcomeCarousel_2_Mobile.png",
                          "downloadUrl": "https://imagestg.deprati.com.ec/sys-master/images/h8e/he8/8969829449758",
                          "mime": "image/png",
                          "url": "https://imagestg.deprati.com.ec/sys-master/images/h8e/he8/8969829449758",
                        },
                        "modifiedTime": "2023-01-25T14:25:41+0000",
                        "name": "Example2 Home Banner Mobile",
                        "typeCode": "BannerComponent",
                        "uid": "WelcomeCarousel_2_Mobile",
                        "urlLink": "/",
                      },
                      Object {
                        "container": "false",
                        "content": "Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit. Hola.",
                        "external": "false",
                        "headline": "Solicita tu Crédito De Prati de forma facíl y rápida - Prueba iPhone",
                        "media": Object {
                          "code": "WelcomeCarousel_3_Mobile.png",
                          "downloadUrl": "https://imagestg.deprati.com.ec/sys-master/images/h23/he8/8969829482526",
                          "mime": "image/png",
                          "url": "https://imagestg.deprati.com.ec/sys-master/images/h23/he8/8969829482526",
                        },
                        "modifiedTime": "2023-01-25T14:25:41+0000",
                        "name": "Example3 Home Banner Mobile",
                        "typeCode": "BannerComponent",
                        "uid": "WelcomeCarousel_3_Mobile",
                        "urlLink": "/",
                      },
                    ]
                  }
                  getItem={[Function]}
                  getItemCount={[Function]}
                  handlerTag={1}
                  handlerType="NativeViewGestureHandler"
                  horizontal={true}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onEndReached={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  pagingEnabled={true}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  renderScrollComponent={[Function]}
                  scrollEventThrottle={50}
                  showsHorizontalScrollIndicator={false}
                  stickyHeaderIndices={Array []}
                  viewabilityConfigCallbackPairs={Array []}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                      style={
                        Array [
                          Object {
                            "flexDirection": "row",
                          },
                          null,
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "width": 675,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#263238",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 18,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "paddingHorizontal": "20%",
                              "paddingTop": "3%",
                              "textAlign": "center",
                            }
                          }
                        >
                          Bienvenido A De Prati App - Prueba
                        </Text>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "height": "60%",
                              "justifyContent": "center",
                              "marginTop": 20,
                              "width": "85%",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "uri": "https://imagestg.deprati.com.ec/sys-master/images/h6f/h47/8969902358558",
                              }
                            }
                            style={
                              Object {
                                "height": "100%",
                                "resizeMode": "contain",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginTop": 15,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#263238",
                                "fontFamily": "Roboto-Regular",
                                "fontSize": 14,
                                "fontStyle": "normal",
                                "fontWight": "400",
                                "letterSpacing": 0.25,
                                "paddingHorizontal": "10%",
                                "textAlign": "center",
                              }
                            }
                          >
                            Prueba iPhone - texto texto texto
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={
                        Array [
                          Object {
                            "flexDirection": "row",
                          },
                          null,
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "width": 675,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#263238",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 18,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "paddingHorizontal": "20%",
                              "paddingTop": "3%",
                              "textAlign": "center",
                            }
                          }
                        >
                          Encuentra tus productos favoritos
                        </Text>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "height": "60%",
                              "justifyContent": "center",
                              "marginTop": 20,
                              "width": "85%",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "uri": "https://imagestg.deprati.com.ec/sys-master/images/h8e/he8/8969829449758",
                              }
                            }
                            style={
                              Object {
                                "height": "100%",
                                "resizeMode": "contain",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginTop": 15,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#263238",
                                "fontFamily": "Roboto-Regular",
                                "fontSize": 14,
                                "fontStyle": "normal",
                                "fontWight": "400",
                                "letterSpacing": 0.25,
                                "paddingHorizontal": "10%",
                                "textAlign": "center",
                              }
                            }
                          >
                            Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={
                        Array [
                          Object {
                            "flexDirection": "row",
                          },
                          null,
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "justifyContent": "center",
                            "width": 675,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#263238",
                              "fontFamily": "Roboto-Regular",
                              "fontSize": 18,
                              "fontStyle": "normal",
                              "fontWeight": "500",
                              "paddingHorizontal": "20%",
                              "paddingTop": "3%",
                              "textAlign": "center",
                            }
                          }
                        >
                          Solicita tu Crédito De Prati de forma facíl y rápida - Prueba iPhone
                        </Text>
                        <View
                          style={
                            Object {
                              "alignSelf": "center",
                              "height": "60%",
                              "justifyContent": "center",
                              "marginTop": 20,
                              "width": "85%",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "uri": "https://imagestg.deprati.com.ec/sys-master/images/h23/he8/8969829482526",
                              }
                            }
                            style={
                              Object {
                                "height": "100%",
                                "resizeMode": "contain",
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginTop": 15,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#263238",
                                "fontFamily": "Roboto-Regular",
                                "fontSize": 14,
                                "fontStyle": "normal",
                                "fontWight": "400",
                                "letterSpacing": 0.25,
                                "paddingHorizontal": "10%",
                                "textAlign": "center",
                              }
                            }
                          >
                            Neque porro quisquam est, qui dolorem ipsum, quia dolor sit, amet, consectetur, adipisci velit. Hola.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "flexDirection": "row",
                      "height": "5%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderRadius": 5,
                          "height": 10,
                          "marginHorizontal": 5,
                          "width": 10,
                        },
                        Object {
                          "backgroundColor": "#E31A2C",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "borderRadius": 5,
                          "height": 10,
                          "marginHorizontal": 5,
                          "width": 10,
                        },
                        Object {
                          "backgroundColor": "#E6E6E6",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "borderRadius": 5,
                          "height": 10,
                          "marginHorizontal": 5,
                          "width": 10,
                        },
                        Object {
                          "backgroundColor": "#E6E6E6",
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": -50,
                    "height": 300,
                    "paddingTop": 20,
                    "width": "100%",
                  }
                }
              >
                <View
                  accessibilityLabel="welcome_iniciar_sesion"
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E31A2C",
                      "borderRadius": 5,
                      "justifyContent": "center",
                      "opacity": 1,
                      "paddingVertical": 10,
                      "width": "95%",
                    }
                  }
                  testID="welcome_iniciar_sesion"
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": undefined,
                          },
                          Object {},
                          Object {},
                        ]
                      }
                    >
                      INICIAR SESIÓN
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityLabel="welcome_btn_omitir"
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "opacity": 1,
                      "paddingVertical": 14,
                    }
                  }
                  testID="welcome_btn_omitir"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#263238",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 14,
                          "fontWeight": undefined,
                          "letterSpacing": 0.6,
                          "textDecorationLine": "underline",
                        },
                        Object {},
                        Object {},
                      ]
                    }
                  >
                    Invitado
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
